package.json

```diff
@@ -69,7 +69,7 @@
     "semver": "^7.5.4",
     "ts-jest": "^28.0.7",
     "ts-node": "^10.7.0",
-    "typescript": "~4.8.4"
+    "typescript": "~5.0.4"
```

Comment:
Similar question as for https://github.com/MetaMask/abi-utils/pull/73: Are we upgrading to TypeScript just to verify that we are able to switch to `moduleResolution` = `nodenext` and therefore ensure that the upgrade to `@metamask/superstruct` doesn't produce the same errors that we were seeing in `core`? In other words, is this upgrade required to use `@metamask/superstruct`? If it's not, should we do it in a different PR?

Reply:
The typescript version bump isn't required for enabling `nodenext`. We can definitely merge the compiler options changes first.

But since we're not seeing any regressions in CI, it seemed safe to include the typescript upgrade here, especially given it's a devDep.

Reply:
Hmm, I guess upgrading this dependency and upgrading TypeScript seem like separate concerns to me, so maybe we ought to have them in separate PRs so they're easier to trace later if need be?

Reply:
Reverted the typescript bumps here and in the other open PRs:
https://github.com/MetaMask/eth-snap-keyring/pull/311
https://github.com/MetaMask/keyring-api/pull/328
https://github.com/MetaMask/abi-utils/pull/73